internal/server/v1/firehose/mappings.go

```diff
@@ -160,19 +145,18 @@ func makeConfigStruct(cfg *models.FirehoseConfig, prj *shieldv1beta1.Project) (*
 	cfg.EnvVars["STREAM_NAME"] = *cfg.StreamName
 	cfg.EnvVars["INPUT_SCHEMA_PROTO_CLASS"] = *cfg.InputSchemaProtoClass
 
-	return utils.GoValToProtoStruct(moduleConfig{
-		State:    "RUNNING",
-		StopTime: stopAt,
-		Telegraf: telegrafConf,
-		Firehose: moduleConfigFirehoseDef{
-			Replicas:           int(*cfg.Replicas),
-			KafkaBrokerAddress: *cfg.BootstrapServers,
-			KafkaTopic:         *cfg.TopicName,
-			KafkaConsumerID:    *cfg.ConsumerGroupID,
-			EnvVariables:       cfg.EnvVars,
-			DeploymentID:       cfg.DeploymentID,
-		},
-	})
+	var entropyFirehoseConfig entropyFirehose.Config
+	entropyFirehoseConfig.State = "RUNNING"
+	entropyFirehoseConfig.StopTime = stopAt
+	entropyFirehoseConfig.Telegraf = telegrafConf
+	entropyFirehoseConfig.Firehose.Replicas = int(*cfg.Replicas)
+	entropyFirehoseConfig.Firehose.KafkaBrokerAddress = *cfg.BootstrapServers
+	entropyFirehoseConfig.Firehose.KafkaTopic = *cfg.TopicName
+	entropyFirehoseConfig.Firehose.KafkaConsumerID = *cfg.ConsumerGroupID
+	entropyFirehoseConfig.Firehose.EnvVariables = cfg.EnvVars
+	entropyFirehoseConfig.Firehose.DeploymentID = cfg.DeploymentID
```

Comment:
use the inlined struct init as before? (easier to manage a single struct init statement)

Reply:
since the Firehose struct definition is nested, it isn't public. how can I init the firehose struct then?

Reply:
oh damn. you are right. leave it like this for now. i want to un-nest this struct as well.. this nesting doesn't make any sense. but for now keep it that way.